zio/zstio/reader.go

```diff
@@ -0,0 +1,25 @@
+package zstio
+
+import (
+	"errors"
+	"io"
+
+	"github.com/brimsec/zq/zng/resolver"
+	"github.com/brimsec/zq/zst"
+)
+
+type Reader struct {
+	zst.Reader
+}
+
+func NewReader(r io.Reader, zctx *resolver.Context) (*Reader, error) {
+	seeker, ok := r.(zst.Seeker)
+	if !ok {
+		return nil, errors.New("zst must be used with a seekable input")
+	}
+	reader, err := zst.NewReaderFromSeeker(zctx, seeker)
+	if err != nil {
+		return nil, err
+	}
+	return &Reader{*reader}, nil
```

Comment:
Nit: This dereference is a little odd. Is there any reason the embedded zst.Reader field shouldn't just be a *zst.Reader?

Reply:
Because it wraps the struct rather than the interface... it makes sense to me.  Why have an extra pointer when you don't need?

Reply:
The difference is that the zst reader implementation lives in zq/zst.  Perhaps it should move to zstio then there is only one reader and zst.Object/Assembler live in zq/zst?

Reply:
Hmm, the dissonance here is that I'm wrapping the struct when I don't have to and can just return a zst.Reader (and zst.Writer) since they implement zbuf.Reader (and zbuf.WriteCloser).  I'll fix.